test/IDE/reconstruct_type_from_mangled_name.swift

```diff
@@ -129,10 +168,101 @@ func test001() {
 // CHECK: type: String
   _ = genstruct2.z
 // CHECK: type: Dictionary<Int, Int>
+
+  genstruct2.takesT(123)
 }
 
+// CHECK: decl: protocol P1
 protocol P1 {}
-func foo1(p : P1) {}
-// CHECK: decl: protocol P1  for 'P1' usr=s:14swift_ide_test2P1P
-// CHECK: decl: func foo1(p: P1)  for 'foo1' usr=s:14swift_ide_test4foo1yAA2P1_p1p_tF
-// CHECK: decl: let p: P1 for 'p' usr=s:14swift_ide_test4foo1yAA2P1_p1p_tFADL_AaC_pv
+
+// CHECK: decl: func foo1(p: P1)
+func foo1(p: P1) {
+// CHECK: decl: let p: P1
+// CHECK: type: (P1) -> ()
+  foo1(p: p)
+}
+
+// CHECK: decl: protocol P2
+protocol P2 {}
+
+// CHECK: decl: func foo2(p: P1 & P2)
+func foo2(p: P1 & P2) {
+// CHECK: decl: let p: P1 & P2
+  foo2(p: p)
+}
+
+// CHECK: func foo3(p: AnyObject & P1)
+func foo3(p: P1 & AnyObject) {
+// CHECK: decl: let p: AnyObject & P1
+// CHECK: dref: FAILURE for 'AnyObject'
```

Comment:
I haven't committed any of those changes yet, so it's definitely your fault. :-)